test/gpu/native/studies/transpose/transpose.chpl

```diff
@@ -3,15 +3,59 @@ use GpuDiagnostics;
 use GPU;
 use Time;
 
+enum implType {
```

Comment:
If you want to make this more documenting you could add a one sentence description in comments for what each of these different implementations are.

Did you base this code off this blog post:

https://developer.nvidia.com/blog/efficient-matrix-transpose-cuda-cc/

If so you could could just cite that too and say the naive version is the naive version from that and the clever and lowlevel ones are the coalesced version where lowlevel launches using primitives and clever doesn't.

Reply:
I didn't use the blog post, but some sample code from Nvidia. I don't mind citing the blog post, though, and using it for documentation.